File: 1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.cpp
class Solution {
public:
    int numOfSubarrays(vector<int>& arr, int k, int threshold) {
        int count = 0;
        
        int curSum = 0;
        int left = 0;
        for(int right = 0; right < arr.size(); right++) {
            curSum+= arr[right];
            
            if(right-left +1 == k) {
                int avg = curSum / k;
                
                if(avg >= threshold) {
                    count++;
                }
                curSum -= arr[left];
                left++;
            }
        }
        
        return count;
    }
};

File: 1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.py
class Solution:
    def numOfSubarrays(self, arr: List[int], k: int, threshold: int) -> int:
        count = 0
        left = 0
        curSum = 0
        
        for right in range(len(arr)):
            curSum += arr[right]
            
            if right-left+1 == k:
                avg = curSum / k
                if avg >= threshold:
                    count+=1
                curSum -= arr[left]
                left+=1
        
        return count;